src/api.cc

```diff
@@ -58,17 +58,29 @@ const std::vector<boost::filesystem::path> AkliteClient::CONFIG_DIRS = {"/usr/li
 // It is applicable only to offline/local update.
 const static std::string LocalSrcDirKey{"local-src-dir"};
 
-TufTarget CheckInResult::GetLatest(std::string hwid) const {
+TufTarget CheckInResult::GetLatest(std::string hwid) const { return SelectTarget(-1, "", std::move(hwid)); }
+
+TufTarget CheckInResult::SelectTarget(int version, const std::string& target_name, std::string hwid) const {
   if (hwid.empty()) {
     hwid = primary_hwid_;
   }
 
-  for (auto it = targets_.crbegin(); it != targets_.crend(); ++it) {
-    if ((*it).Custom()["hardwareIds"][0] == hwid) {
-      return *it;
+  if (version == -1 && target_name.empty()) {
+    for (auto it = targets_.crbegin(); it != targets_.crend(); ++it) {
+      if ((*it).Custom()["hardwareIds"][0] == hwid) {
+        return *it;
+      }
+    }
+  } else {
+    for (const auto& t : targets_) {
+      if (t.Version() == version || t.Name() == target_name) {
+        return t;
+      }
     }
   }
-  throw std::runtime_error("no target for this hwid");
+
+  LOG_INFO << "no target for hwid " << hwid;
+  return TufTarget();
```

Comment:
Replacing the exception with returning of empty target may cause regression.
Is there particular need to return an empty target if not found?

Reply:
I'm re-adding the `throw` statement to the `GetLatest` method, so that users of the original method will still have the original behavior.
I prefer to keep `SelectTarget` returning empty instead of throwing an exception if no matching target is found. It may be just a matter of a given target name not being in the list, which is a regular situation, not an exception. And the caller code gets also a little cleaner.
But is easy to change that if we believe it is better to have an exception. I've reverted the original `GetLatest` calls back to the original name, so there is only two places where `SelectTarget` is currently used: inside `GetLatest` and `GetTargetToInstall`.

Reply:
I don't have strong opinion on it. The current version LGTM.